1.MLP_Binary_Classifier/nbconverted/Hyperparameter_Optimization_binary_class.py

```diff
@@ -0,0 +1,248 @@
+
+params = Parameters()
+
+
+# In[2]:
+
+
+# Import Data
+# set data file path under pathlib path for multi-system use
+file_path = Path(
+    "../../Extracted_Features_(CSV_files)/interstellar_wave3_sc_norm_cellprofiler.csv.gz"
+)
+df = pd.read_csv(
+    file_path,
+    low_memory=False,
+)
+
+
+# Combine treatment with dosage to be able to discern treatments with different doses as a different condition
+
+# In[3]:
+
+
+# Combine treatment and dose
+df["Metadata_treatment"] = df["Metadata_treatment"] + "_" + df["Metadata_dose"]
+print(df["Metadata_treatment"].unique())
+
+# Generate df speceific to analysis and model
+df = df.query(
+    "Metadata_treatment == 'LPS_10µg/ml'| Metadata_treatment == 'Media only_0'"
+)
+print(df["Metadata_treatment"].unique())
+
+df_stats(df)
+# Drop na and reindex accordingly
+df = df.dropna()
+df.reindex()
+# Check for Nans again
+df_stats(df)
+# Understand categorical data such as treatment and dosing
+df[["Metadata_treatment", "Metadata_dose"]].drop_duplicates()
+if params.SUBSET_OPTION:
+    df = df.sample(n=params.SUBSET_NUMBER)
+else:
+    pass
+# Code snipptet for metadata extraction by Jenna Tomkinson
+df_metadata = list(df.columns[df.columns.str.startswith("Metadata")])
+
+# define which columns are data and which are descriptive
+df_descriptive = df[df_metadata]
+df_values = df.drop(columns=df_metadata)
+
+
+# ### Setting up data for network training
+
+# In[4]:
+
+
+# Creating label encoder
+le = preprocessing.LabelEncoder()
+# Converting strings into numbers
+df_values["Metadata_treatment"] = le.fit_transform(df_descriptive["Metadata_treatment"])
+# split into X and Y where Y are the predictive column and x are the observable data
+df_values_X = df_values.drop("Metadata_treatment", axis=1)
+df_values_Y = df_values["Metadata_treatment"]
+
+X_train, X_test, X_val, Y_train, Y_test, Y_val = data_split(
+    df_values_X, df_values_Y, 0.8, 0.1, 0.1
+)
+
+
+# In[5]:
+
+
+# produce data objects for train, val and test datasets
+train_data = Dataset(
+    torch.FloatTensor(X_train.values), torch.FloatTensor(Y_train.values)
+)
+val_data = Dataset(torch.FloatTensor(X_val.values), torch.FloatTensor(Y_val.values))
+test_data = Dataset(torch.FloatTensor(X_test.values), torch.FloatTensor(Y_test.values))
+
+IN_FEATURES = X_train.shape[1]
+print("Number of in features: ", IN_FEATURES)
+out_features = len(df_values["Metadata_treatment"].unique())
+print("Number of out features: ", out_features)
+
+
+# In[6]:
+
+
+# convert data class into a dataloader to be compatible with pytorch
+train_loader = torch.utils.data.DataLoader(
+    dataset=train_data, batch_size=params.BATCH_SIZE
+)
+valid_loader = torch.utils.data.DataLoader(
+    dataset=val_data, batch_size=params.BATCH_SIZE
+)
+test_loader = torch.utils.data.DataLoader(
+    dataset=test_data, batch_size=params.BATCH_SIZE
+)
+
+
+# In[7]:
+
+
+# wrap the objective function inside of a lambda function to pass args...
+objective_lambda_func = lambda trial: objective(
```

Comment:
consider using named function arguments, rather than position-based arguments. It's so much easier to read with named functions!